src/logic/safe/hooks/useLoadSafe.tsx

```diff
@@ -15,6 +16,7 @@ export const useLoadSafe = (safeAddress?: string): boolean => {
   useEffect(() => {
     const fetchData = async () => {
       if (safeAddress) {
+        await dispatch(fetchSelectedCurrency())
```

Comment:
I see there's a dispatch inside `fetchSelectedCurrency`, does it need another dispatch on top?

Reply:
Yes, this is how thunks work https://redux.js.org/tutorials/fundamentals/part-6-async-logic

Reply:
🤷‍♂️

Reply:
Should i do anything else for this to go into QA?

Reply:
Approved. I was just wondering why it has a second dispatch.